packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable i18next/no-literal-string */
+import { 
+  Component, 
+  Element, 
+  Host, 
+  h, 
+  Prop, 
+  forceUpdate,
+  Fragment,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+import i18next from 'i18next';
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the file input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button.
+   */
+  @Prop() buttonText: string;
+
+  /**
+   * Sets the input to required and renders the (*Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  /**
+   * Makes sure the button text always has a value.
+   * @returns string - Button text to render.
+   */
+  private getButtonText = (): string => {
+    return this.buttonText ? this.buttonText : 'Upload file';
```

Comment:
I'm still not sure why we need this function.

Reply:
See my comment here: https://github.com/department-of-veterans-affairs/component-library/pull/507#discussion_r948267986